### PR TITLE
Add models with validation

### PR DESCRIPTION
**Pull request by sujanadiga**
### What ?
- Add article and comments models
- CRUD operations on article model
- Creare and Destroy actions on comment model
- UI enabling CRU on article model
### Why ?
- To create a blog application

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  def show
+    @article = Article.find(params[:id])
+  end
+
+  def edit
+    @article = Article.find(params[:id])
+  end
+
+  def destroy
+    @article = Article.find(params[:id])
```
Comment (tarunjadhwani) [inline app/controllers/articles_controller.rb]: DRY your code by removing this line into a function

File: app/models/article.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Article < ActiveRecord::Base
+  has_many :comments, dependent: :destroy
+
+  validates :title, presence: true
```
Comment (tarunjadhwani) [inline app/models/article.rb]: club it. DRY code.

File: app/models/article.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Article < ActiveRecord::Base
+  has_many :comments, dependent: :destroy
```
Comment (tarunjadhwani) [inline app/models/article.rb:2]: Look for other options available for dependents

File: config/routes.rb [hunk 4]
```diff
@@ -53,4 +53,8 @@
   #     # (app/controllers/admin/products_controller.rb)
   #     resources :products
   #   end
+  resources :articles do
+    resources :comments
```
Comment (tarunjadhwani) [inline config/routes.rb]: check "accept" and "only". Add only relevant routes

**Review by tarunjadhwani — CHANGES_REQUESTED** [review]

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (sujanadiga) [reply app/controllers/articles_controller.rb]: Created a private function for the same

File: app/models/article.rb (hunk 3 above)
Comment (sujanadiga) [reply app/models/article.rb:2]: Other options are,
- delete_all(no callbacks called - database deletion)
- nullify (make foreign keys NULL)
- restrict_with_exception|error (restrict with exception or error)

File: app/models/article.rb (hunk 2 above)
Comment (sujanadiga) [reply app/models/article.rb]: Clubbed validates together

File: config/routes.rb (hunk 4 above)
Comment (sujanadiga) [reply config/routes.rb]: generating routes for only 2 actions now(destroy and create)

**Review by tarunjadhwani — APPROVED** [review]

**Pull request merged**